lib/zlib.js

```diff
@@ -173,14 +174,13 @@ function zlibOnError(message, errno, code) {
   const self = this[owner_symbol];
   // There is no way to cleanly recover.
   // Continuing only obscures problems.
-  _close(self);
-  self._hadError = true;
 
   // eslint-disable-next-line no-restricted-syntax
   const error = new Error(message);
   error.errno = errno;
   error.code = code;
-  self.emit('error', error);
+  self.destroy(error);
+  self._error = error;
```

Comment:
Is this an already present property? If not, let's just use a symbol instead?

Reply:
Fixed, uses symbol